### DISCUSSION TITLE
question: 元件繼承問題

### DISCUSSION THREAD

**Discussion by a0926019412** [dotnetcore/BootstrapBlazor#2467] [Q&A] [answered]
您好 想請教元件繼承問題
因為想要在部分元件在使用時直接給預設值 , 想繼承select元件並給予一些預設值 (比如ShowSearch) 
下圖為razor程式碼

![inherit](https://github.com/dotnetcore/BootstrapBlazor/assets/4286244/e37004fd-fcb0-4d47-9082-2ef8640eab61)

但使用後會報以下錯誤
![image](https://github.com/dotnetcore/BootstrapBlazor/assets/4286244/7306858b-413a-49d2-8a37-d4392973afca)

請問是我用法錯誤嗎?

**Comment by bb-auto** [comment]
@a0926019412 Thank you for reporting. We will give triage later.

**Comment by ArgoZhang** [comment, marked as answer]
@a0926019412 组件是需要 js 工作的。你需要设置脚本路径

`@attribute [JSModuleAutoLoader("./_content/xxxxx/Select.razor.js", JSObjectReference = true)]`

**Comment by ArgoZhang** [comment]
@a0926019412 其实最好不要继承而是包裹

razor file
```
<Select ShowSearch="true" Items="@Items"></Select>
```

**Comment by a0926019412** [comment]
嗨 argo

因為想把select 組件本身的attribute也想 open給其他元件調整, 又懶得一條一條開出來, 因此想繼承後再去擴充給內部使用
@attribute [JSModuleAutoLoader("./_content/xxxxx/Select.razor.js", JSObjectReference = true)]
我會再試試看 謝謝支援
